astropy/timeseries/periodograms/lombscargle/_statistics.py

```diff
@@ -342,36 +344,40 @@ def inv_fap_baluev(p, fmax, t, y, dy, normalization='standard'):
     return res.x
 
 
-def _bootstrap_max(t, y, dy, fmax, normalization, random_seed):
+def _bootstrap_max(t, y, dy, fmax, normalization, random_seed, n_bootstrap=1000):
     """Generate a sequence of bootstrap estimates of the max"""
     from .core import LombScargle
     rng = np.random.RandomState(random_seed)
-    while True:
+    power_max = []
```

Comment:
It may be an idea to pre-allocate the array. For instance,
```
    power_max = np.empty(n_bootstrap, float)
    for i in range(n_bootstrap):
        s = rng.randint(0, len(y), len(y))  # sample with replacement
        ls_boot = LombScargle(t, y[s], dy if dy is None else dy[s],
                              normalization=normalization)
        freq, power = ls_boot.autopower(maximum_frequency=fmax)
        if i == 0 and hasattr(power, 'unit'):
            power_max = u.Quantity(power_max, power.unit, copy=False)
        power_max[i] = power.max()
```

Reply:
Working with the list approach seems a tiny bit less slow, but arguably I didn't run proper benchmarking only a few %timeit runs, so I suppose ultimately they are the same.

Reply:
Hmm, I actually found the same but assumed it was just a coincidence. Sounds like assigning to an array is slower than appending to a list.

The one advantage of my suggestion is that it will return an array when no units were used - perhaps a bit less surprising.

Anyway, as said, fine to just keep what you have!

Reply:
oh, of course, I didn't cover that original case to convert the list back to an array. 🤦‍♀

Reply:
as it stands it doesn't really matter whether this private function returns array or a dimensionless Quantity, at the user end it seems to be handled appropriately (the np.searchsort deals with it nicely, etc.).